Updates

loopyd opened the pull request:
fixes:  source encoding problem - for irregular utf-8 codes in output  - causes fatal crash
update:  output cleanup, with error and success counters

loopyd committed:
fixes

source encoding problem - for irregular utf-8 codes in output
output cleanup, with error and success counters

loopyd committed:
minor output fix

rushilsrivastava commented:
Hi, I the solution proposed works for Python 3, but throws errors with Python 2. Also, I would like to keep the current output method, as it allows for quick identification of errors, indicated by `[!]`. 

Could you please add a commit with the requested changes?

rushilsrivastava commented on google_scraper.py:
```diff
@@ -29,24 +29,24 @@ def search(url):
     browser.set_window_size(1024, 768)
     print("\n===============================================\n")
-    print("[%] Successfully launched Chrome Browser")
+    print("==> Chrome Driver Launched")
```
Let's keep the original `[%]` and make `Chrome Driver` -> `Chromedriver`

loopyd replied:
ok, addressing all these fixes now.

rushilsrivastava commented on google_scraper.py:
```diff
@@ -29,24 +29,24 @@ def search(url):
     browser.get(url)
     time.sleep(1)
-    print("[%] Successfully opened link.")
+    print("==> Link Loaded")
```
`[%] Loaded Link`

rushilsrivastava commented on google_scraper.py:
```diff
@@ -29,24 +29,24 @@ def search(url):
     element = browser.find_element_by_tag_name("body")
 
-    print("[%] Scrolling down.")
+    print("==> Scrolling page...")
```
`[%] Scrolling Page`

rushilsrivastava commented on google_scraper.py:
```diff
@@ -29,24 +29,24 @@ def search(url):
     try:
         browser.find_element_by_id("smb").click()
-        print("[%] Successfully clicked 'Show More Button'.")
+        print("==> Generated 'Show More' event")
```
`[%] Loaded Show More Element`

rushilsrivastava commented on google_scraper.py:
```diff
@@ -55,22 +55,26 @@ def search(url):
             time.sleep(0.3)  # bot id protection
 
-    print("[%] Reached end of Page.")
+    print("==> Reached end of page")
```
`[%] Reached the end of the Page`

rushilsrivastava commented on google_scraper.py:
```diff
@@ -55,22 +55,26 @@ def search(url):
 
     time.sleep(1)
     # Get page source and close the browser
+    print("==> Saving page source...")
```
Would rather just have 1 line for the source.

rushilsrivastava commented on google_scraper.py:
```diff
@@ -55,22 +55,26 @@ def search(url):
     source = browser.page_source
-    f = open('dataset/logs/google/source.html', 'w+')
-    f.write(source)
+    with open('dataset/logs/google/source.html', 'w+', encoding='utf-8', errors='replace') as f:
```
Works for Python 3. Python 2 Solution can be implemented using the `io` module.

Python 2: https://docs.python.org/2/library/io.html
Python 3: https://docs.python.org/3/library/io.html

rushilsrivastava commented on google_scraper.py:
```diff
@@ -55,22 +55,26 @@ def search(url):
+    with open('dataset/logs/google/source.html', 'w+', encoding='utf-8', errors='replace') as f:
+        f.write(source)
 
+    print("==> Page source saved")
```
`[%] Saved Page Source`

rushilsrivastava commented on google_scraper.py:
```diff
@@ -55,22 +55,26 @@ def search(url):
+	
     browser.close()
-    print("[%] Closed Browser.")
+    print("==> Closed browser window")
```
`[%] Closed Chromedriver`

rushilsrivastava commented on google_scraper.py:
```diff
@@ -55,22 +55,26 @@ def search(url):
 
 def error(link):
-    print("[!] Skipping {}. Can't download or no metadata.\n".format(link))
+    print("==> Skipped. Can't download or no metadata.\n")
```
I would want to keep the link for reference purposes. Since this is logged as an error (`[!]`) users can iterate the links to manually download the images.

rushilsrivastava commented on google_scraper.py:
```diff
@@ -98,7 +102,7 @@ def download_image(link, image_data):
 
         # Download the image
-        print("[%] Downloading Image #{} from {}".format(
+        print("==> Downloading Image #{} from {}".format(
```
`[%]`

rushilsrivastava commented on google_scraper.py:
```diff
@@ -110,11 +114,13 @@ def download_image(link, image_data):
             download_image.delta -= 1
-            print("[!] Issue Downloading: {}\n[!] Error: {}".format(link, e))
+            download_image.errors += 1
+            print("==> Error: {}".format(e))
```
`[!]`

I also think it is important to mention the error and link here.

rushilsrivastava commented on google_scraper.py:
```diff
@@ -110,11 +114,13 @@ def download_image(link, image_data):
         download_image.delta -= 1
-        print("[!] Issue getting: {}\n[!] Error:: {}".format(link, e))
+        download_image.errors += 1
+        print("==> Error: {}".format(e))
```
`[!]` Same as above. It is important to mention the error and link here.

rushilsrivastava commented on google_scraper.py:
```diff
@@ -155,38 +161,41 @@ def download_image(link, image_data):
             title = ""
         link = rg_meta["ou"]
-        print("\n[%] Getting info on: {}".format(link))
+        print("==> Getting info on: {}".format(link))
```
`[%]`

rushilsrivastava commented on google_scraper.py:
```diff
@@ -155,38 +161,41 @@ def download_image(link, image_data):
         except Exception as e:
             images[link] = image_data
-            print("[!] Issue getting data: {}\n[!] Error: {}".format(rg_meta, e))
+            print("==> Issue getting data: {}\n[!] Error: {}".format(rg_meta, e))
```
`[!]`

rushilsrivastava commented on google_scraper.py:
```diff
@@ -155,38 +161,41 @@ def download_image(link, image_data):
+            print("==> Issue getting data: {}\n[!] Error: {}".format(rg_meta, e))
+            errorcounter += 1
 
+        print("==> Information received for: {}".format(link))
```
`[%] Downloaded information for: {}`

rushilsrivastava commented on google_scraper.py:
```diff
@@ -155,38 +161,41 @@ def download_image(link, image_data):
 
     # Open i processes to download
-    print("\n------------------------------------------\n")
+    print ("\n Done fetching metadata.  {} succeeded | {} failed".format(linkcounter,errorcounter))
```
`print("\nDone loading metadata for {} images. Failed to load metadata for {} images.".format(linkcounter, errorcounter))`

rushilsrivastava commented on google_scraper.py:
```diff
@@ -155,38 +161,41 @@ def download_image(link, image_data):
             error(link)
 
-    print("\n\n[%] Done. Downloaded {} images.".format(download_image.delta))
+    print("\n\n[%] Done. Downloaded images.   {} succeeded | {} failed".format(download_image.delta,download_image.errors))
```
`print("\n\n[%] Done. Downloaded {} images. Failed to download {} images.".format(download_image.delta, download_image.errors))`

rushilsrivastava commented on google_scraper.py:
```diff
@@ -74,7 +74,7 @@ def search(url):
 
 def error(link):
-    print("==> Skipped. Can't download or no metadata.\n")
+    print("==> Skipped. Can't download or no metadata.")
```
`[!]`
Like the others, important to have the link for iteration purposes.

rushilsrivastava commented on google_scraper.py:
```diff
@@ -102,14 +102,14 @@ def download_image(link, image_data):
 
         # Download the image
-        print("==> Downloading Image #{} from {}".format(
+        print("\n==> Downloading Image #{} from {}".format(
```
`[%]`

rushilsrivastava commented on google_scraper.py:
```diff
@@ -102,14 +102,14 @@ def download_image(link, image_data):
             else:
                 urllib.urlopen(link, "dataset/google/{}/".format(query) + "Scrapper_{}.{}".format(str(download_image.delta), type))
-            print("[%] Downloaded File")
+            print("==> Downloaded image")
```
`[%]`

loopyd committed:
Fixes:

- Travis-CLI build error for Python 2.7 fixed (improper file.open() syntax) fixed
- Minor output fixes (continued...)

New features:

- --hash and --unique command line switches for google_scraper.py for more options and portability.

loopyd committed:
Requested in pull request (output changes)

loopyd committed:
requirement - openssl needed satisfied

loopyd committed:
more fix for travis-cli

loopyd committed:
fixed - hashlib "unsupported hash type" in travis-cli

loopyd committed:
"invalid keyword argument" fix in travis-cli for 2.7

loopyd commented:
Please wait, I have execution success for 2.7 and 3, recommended we also implement --limit argument for download_image.delta so that we can reduce travis-cli build time for testing the additional command-line arguments.

loopyd committed:
current directory fix for 2.7

current directory fix for python 2.7 across linux and windows - improper relative path resolution.

loopyd committed:
current_directory seperation

current_directory into its own function definition, working on relative paths across operating system and python version.

loopyd commented:
Working on the way that the program handles relative paths to be multiplatform.  (2.7 is experiencing an additional issue in relation to relative path construction)

loopyd committed:
Debug commit

loopyd committed:
Debug commit

loopyd committed:
better errors

- better error handler implimentation
- multiplatform relative paths across all functions

rushilsrivastava commented:
Hi I appreciate the extra changes, however this changes the original scope of the PR. Please open another PR for other features/changes. As per your title, this PR is addressing issues in #5.

rushilsrivastava commented:
I have added a limit argument in both the bing and google scraper in https://github.com/rushilsrivastava/image-scrapers/commit/86effe9d6246756d34b50f50fed030aba54cd6d7
